pkg/api/v1/types.go

```diff
@@ -1444,6 +1444,10 @@ const (
 	// external load balancer (if the cloud provider supports it), in addition
 	// to 'NodePort' type.
 	ServiceTypeLoadBalancer ServiceType = "LoadBalancer"
+
+    // ServiceTypeClosed means a service will only be accessible inside the
+    // namespace, via the Cluster IP.
+	ServiceTypeClosed ServiceType = "Closed"
```

Comment:
As we said at the end of last week, I think we want to call this "NamespaceIP" so it is symmetrical with ClusterIP